doc/development/source/orange-demo/orangedemo/OWDataSamplerA.py

```diff
@@ -3,16 +3,20 @@
 import numpy
 
 import Orange.data
-from Orange.widgets import widget, gui
+from Orange.widgets.widget import OWWidget, Input, Output
+from Orange.widgets import gui
 
-class OWDataSamplerA(widget.OWWidget):
+class OWDataSamplerA(OWWidget):
     name = "Data Sampler"
     description = "Randomly selects a subset of instances from the data set"
     icon = "icons/DataSamplerA.svg"
     priority = 10
 
-    inputs = [("Data", Orange.data.Table, "set_data")]
-    outputs = [("Sampled Data", Orange.data.Table)]
+    class Inputs:
+        data = Input("Data", Orange.data.Table)
+
+    class Outputs:
+        sample = Output("Sampled Data", Orange.data.Table)
```

Comment:
For the record, for decorators to work, we could possibly _use_ metaclass' `__prepare__`: https://www.python.org/dev/peps/pep-3115/#example
```py
class WidgetMeta:
    @classmethod
    def __prepare__(mcs, name, bases, **kwargs):
        return mcs.Dict()

    def __new__(mcs, name, bases, attrs):
        ...

    class Dict(dict):
        def __setitem__(self, item, value):
            if item in ('Input', 'Output'):
                # Transform Input/Output class so that their
                # attributes are of Signal type
                value = ...
            super().__setitem__(item, value)